### PR TITLE
Run the benchmark for each backend * config separately, without duplication 

### PR DESCRIPTION
**Pull request by rolfmorel**
Add --backend option and run benchmark one backend*config option at a time

This branch allows for running baseline triton-cpu from the same script alongside all other backends & config options.

### CODE REVIEW

**Comment by rengolin** [comment]
> This branch is meant to merge on top of #30

Then this should be onto `xsmm-main` branch, not `dev-xsmm-v2`

**Comment by rolfmorel** [comment]
Note for later: as the rebased commits ended up as a single commit on xsmm-main, the single significant commit needs to be rebased on what's now xsmm-main.

File: python/tutorials/03-matrix-multiplication-cpu.py [hunk 1]
```diff
@@ -441,15 +441,25 @@ def matmul(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, M: int, N: int, K:
...
-LINE_STYLES = [('blue', '--'), ('blue', '-'), ('green', '--'), ('green', '-')]
-if DATA_TYPE == torch.float8_e5m2:
+BENCHMARK_BACKEND = os.getenv("BENCHMARK_BACKEND")
+if BENCHMARK_BACKEND:
+    assert(BENCHMARK_BACKEND in {"triton-cpu", "triton-xsmm", "torch-cpu-native", "torch-cpu-compile"})
+    LINE_VALS = [BENCHMARK_BACKEND]
+    LINE_NAMES = [BENCHMARK_BACKEND]
+    LINE_STYLES = [('blue', '-')]
+else:
+    assert(False and "no backend specified")
```
Comment (rolfmorel) [inline python/tutorials/03-matrix-multiplication-cpu.py]: FIXME: remove before merge.

File: python/tutorials/03-matrix-multiplication-cpu.py [hunk 2]
```diff
@@ -441,15 +441,25 @@ def matmul(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, M: int, N: int, K:
...
     LINE_VALS = [
-        'triton-cpu-single', 'triton-cpu']
-    LINE_NAMES = ['TritonCPU 1', 'TritonCPU']
-    LINE_STYLES = [('blue', '--'), ('blue', '-')]
+        'triton-cpu-single', 'triton-cpu', 'torch-cpu-native', 'torch-cpu-compile']
+    LINE_NAMES = ['TritonCPU 1', 'TritonCPU', 'TorchCPU (native)', 'TorchCPU (compile)']
+    LINE_STYLES = [('blue', '--'), ('blue', '-'), ('green', '--'), ('green', '-')]
+#LINE_VALS = ['triton-cpu']
```
Comment (rolfmorel) [inline python/tutorials/03-matrix-multiplication-cpu.py]: CLEAN-ME

**Pull request merged**